cylc/flow/scheduler_cli.py

```diff
@@ -417,7 +417,7 @@ async def _run(parser, options, reg, is_restart, scheduler):
     # stop cylc stop
     except SchedulerError:
         ret = 1
-    except KeyboardInterrupt as exc:
+    except (KeyboardInterrupt, asyncio.exceptions.CancelledError) as exc:
```

Comment:
In light of example code and discussion on this issue, we can't catch `KeyboardInterrupt` here at all (whether or not catching `CancelledError` is the right solution).